foundationdb/src/tuple.rs

```diff
@@ -216,6 +216,37 @@ impl Single for String {
     }
 }
 
+impl Single for Vec<SingleValue> {
+    fn encode<W: Write>(&self, w: &mut W) -> std::io::Result<()> {
+        NESTED.write(w)?;
+        for v in self {
+            v.encode(w)?;
+        }
+        EMPTY.write(w)
+    }
+
+    fn decode(buf: &[u8]) -> Result<(Self, usize)> {
+        if buf.len() < 2 {
+            return Err(TupleError::EOF);
+        }
+
+        NESTED.expect(buf[0])?;
+
+        let mut tuples = Vec::new();
+        let mut idx = 1;
+        while buf[idx] != EMPTY {
```

Comment:
We should check index boundary condition. Following code panics with out-of-bound

```
SingleValue::decode(&[NESTED, INTZERO])
```